packages/tpetra/core/ext/TpetraExt_TripleMatrixMultiply_def.hpp

```diff
@@ -492,7 +492,6 @@ namespace Tpetra {
       // Run through all the hash table lookups once and for all
       lo_view_t targetMapToOrigRow(Kokkos::ViewAllocateWithoutInitializing("targetMapToOrigRow"),Aview.colMap->getNodeNumElements());
       lo_view_t targetMapToImportRow(Kokkos::ViewAllocateWithoutInitializing("targetMapToImportRow"),Aview.colMap->getNodeNumElements());
-      Kokkos::fence();
       Kokkos::parallel_for("Tpetra::mult_R_A_P_newmatrix::construct_tables",range_type(Aview.colMap->getMinLocalIndex(), Aview.colMap->getMaxLocalIndex()+1),KOKKOS_LAMBDA(const LO i) {
```

Comment:
@csiefer2 @brian-kelley  Fun fact: these loop bounds work even if the calling process has no indices, but only because `getMaxLocalIndex()` returns `OrdinalTraits<LO>::invalid()` which happens to be `-1` for signed or `numeric_limits<LO>::max()` for unsigned.